src/nupic/engine/Region.cpp

```diff
@@ -684,17 +690,58 @@ void Region::loadDims(std::map<std::string,Dimensions>& outDims,
   }
 }
 
+void Region::getOutputBuffers_(std::map<std::string, Array>& buffers) const {
+	for (auto iter : outputs_) {
+    buffers[iter.first] = iter.second->getData();
+	}
+}
+
+void Region::restoreOutputBuffers_(const std::map<std::string, Array>& buffers) {
+  RegionImplFactory &factory = RegionImplFactory::getInstance();
+  spec_ = factory.getSpec(type_);
+  createInputsAndOutputs_();
+  for (auto output: buffers) {
+    Array& outputBuffer = getOutput(output.first)->getData();
+    outputBuffer = output.second;
+  }
+}
+
+
 void Region::serializeImpl(ArWrapper& arw) const{
     impl_->cereal_adapter_save(arw);
 }
 void Region::deserializeImpl(ArWrapper& arw) {
     RegionImplFactory &factory = RegionImplFactory::getInstance();
-    spec_ = factory.getSpec(type_);
-    createInputsAndOutputs_();
-
     impl_.reset(factory.deserializeRegionImpl(type_, arw, this));
 }
 
+std::ostream &operator<<(std::ostream &f, const Region &r) {
+  f << "Region: {\n";
+  f << "name: " << r.name_ << "\n";
+  f << "nodeType: " << r.type_ << "\n";
+  f << "phases: [ ";
+  for (const auto &phases_phase : r.phases_) {
+      f << phases_phase << " ";
+  }
+  f << "]\n";
+  f << "outputs: [\n";
+  for(auto out: r.outputs_) {
+    f << out.first << " " << out.second->getDimensions() << "\n";
+  }
+  f << "]\n";
+  f << "inputs: [\n";
+  for(auto in: r.inputs_) {
+    f << in.first << " " << in.second->getDimensions() << "\n";
+  }
+  f << "]\n";
+	// TODO: add region impl...maybe
+  //f << "RegionImpl:\n";
+  // Now serialize the RegionImpl plugin.
```

Comment:
this comment seems outdated?

Reply:
> this comment seems outdated?

The whole function will be deleted shortly.